src/dashboards/components/RefreshContext.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {createContext, useReducer, useCallback, FC} from 'react'
+import {useSelector, useDispatch} from 'react-redux'
+import moment from 'moment'
+
+// Types
+import {
+  CustomTimeRange,
+  AutoRefreshStatus,
+  AutoRefresh,
+  AppState,
+} from 'src/types'
+
+// Actions
+import {
+  setAutoRefreshInterval,
+  setAutoRefreshStatus,
+  setAutoRefreshDuration,
+  setInactivityTimeout,
+} from 'src/shared/actions/autoRefresh'
+
+export const AutoRefreshContext = createContext(null)
+
+interface AutoRefreshState {
+  duration: CustomTimeRange
+  inactivityTimeout: string
+  inactivityTimeoutCategory: string
+  refreshMilliseconds: AutoRefresh
+}
+
+const calculateTimeout = (timeout, timeoutUnit) => {
+  const timeoutNumber = parseInt(timeout)
+  const startTime = moment(new Date())
+  const copyStart = startTime.unix()
+  const endTime = startTime.add(timeoutNumber, timeoutUnit[0].toLowerCase())
+  const cutoff = endTime.unix() - copyStart
+  return cutoff * 1000
+}
+
+export const createAutoRefreshInitialState = (
+  override = {}
+): AutoRefreshState => {
+  return {
+    inactivityTimeout: 'None',
+    inactivityTimeoutCategory: 'Hours',
+    duration: {
+      lower: new Date().toISOString(),
+      upper: new Date().toISOString(),
+      type: 'custom',
+    } as CustomTimeRange,
+    refreshMilliseconds: {
+      interval: 0,
+      status: AutoRefreshStatus.Paused,
+    },
+    ...override,
+  }
+}
+
+const autoRefreshReducer = (
+  state = createAutoRefreshInitialState(),
+  action
+) => {
+  switch (action.type) {
+    case 'SET_DURATION':
+      return {...state, duration: action.duration}
+    case 'SET_INACTIVITY_TIMEOUT':
+      return {...state, inactivityTimeout: action.inactivityTimeout}
+    case 'SET_INACTIVITY_TIMEOUT_CATEGORY':
+      return {
+        ...state,
+        inactivityTimeoutCategory: action.inactivityTimeoutCategory,
+      }
+    case 'SET_REFRESH_MILLISECONDS':
+      return {...state, refreshMilliseconds: action.refreshMilliseconds}
+    case 'RESET':
+      return createAutoRefreshInitialState()
+    default:
+      return state
+  }
+}
+
+const AutoRefreshContextProvider: FC = ({children}) => {
+  const [state, dispatch] = useReducer(
+    autoRefreshReducer,
+    createAutoRefreshInitialState()
+  )
+
+  const {currentDashboardId} = useSelector((appState: AppState) => ({
+    currentDashboardId: appState.currentDashboard.id,
+  }))
+
+  const reduxDispatch = useDispatch()
```

Comment:
why isn't this named `dispatch` like every other instance of this pattern in the codebase?

Reply:
https://github.com/influxdata/ui/pull/1226/files#diff-c97616c7c842d06e243a0e48f4b5eed1b608337fce52a4d9bc8bd84d068b3965R82